src/templates/lesson.jsx

```diff
@@ -158,21 +145,24 @@ export const pageQuery = graphql`
               }
             }
           }
-        }
-        three {
-          post {
-            id
-            childMarkdownRemark {
-              fields {
-                slug
-              }
-              frontmatter {
-                title
+          chapters {
+            title
+            entries {
+              entry {
+                id
+                childMarkdownRemark {
+                  fields {
+                    slug
+                  }
+                  frontmatter {
+                    title
+                  }
+                }
               }
             }
           }
         }
       }
     }
   }
```

Comment:
Actually I wanted to this even nicer, but it is not possible.

First read [this](https://github.com/facebook/graphql/issues/91): Arbitrary recursion is not possible within a graphql query.

I tried to make this more DRY like so:

```graphql
{
  tableOfContents: lessonsJson {
     	...ChaptersRecursive
	}
}

fragment ChaptersRecursive on LessonsJson {
  chapters {
    ...chapterFields
    chapters {
      ...chapterFields
      chapters {
        ...chapterFields
      }
    }
  }
}

fragment chapterFields on Chapters {
  title
  entries {
    entry {
      id
      childMarkdownRemark {
        fields {
          slug
        }
        frontmatter {
          title
        }
      }
    }
  }
}
```

But the st***d 'gatsby-transformer-json' creates a type for each individual `chapter` ocurrence instead of one `Chapter` type. You can paste the above in your local graphiql page to see what I mean. One would have to manually set the type of the chapters to a type `Chapter`, but I do not understand how to configure 'gatsby-transformer-json'.